### PR TITLE
Add BigInt and Principal parsing support

### PR DESCRIPTION
**Pull request by earthgohan**
* Update signMessage to sign
* Update enable to connect
* Added support for BigInt and Principal parsing

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,9 +53,13 @@ const injectEarth = () => {
 const handleEarthEnable = () => {
   await injectEarth();
     window?.earth
-    .enable().then((account) => {
+    .connect().then((account) => {
         console.log("Successfully connected to Earth Wallet 🌍", account);
-        onConnect();
+        window.earth.sign({
```
Comment (everhusk) [inline README.md:58]: Why did you add this here? Seems extra

Comment (earthgohan) [reply README.md:58]: `onConnect` doesnt exist at all. So, I have just added a simple sign query

Comment (everhusk) [reply README.md:58]: Got it

**Review by everhusk — APPROVED** [review]
